src/ansi-c/c_typecheck_code.cpp

```diff
@@ -505,7 +505,12 @@ void c_typecheck_baset::typecheck_for(codet &code)
 void c_typecheck_baset::typecheck_label(code_labelt &code)
 {
   // record the label
-  labels_defined[code.get_label()]=code.source_location();
+  if(!labels_defined.emplace(code.get_label(), code.source_location()).second)
+  {
+    error().source_location = code.source_location();
+    error() << "duplicate label '" << code.get_label() << "'" << eom;
```

Comment:
💡 Can this be substituted for a `throw` of an `invalid_source_file_exceptiont` from `src/utils/exception_utils.h`? That would be great!

Reply:
I'm going to do that once #5835 and #5837 have been merged.